server.go

```diff
@@ -341,3 +343,32 @@ func Close() {
 	servers = nil
 	m.Unlock()
 }
+
+// gracefulHandler is used by GracefulServer to prevent calling ServeHTTP on
+// to be closed kept-alive connections during the server shutdown.
+type gracefulHandler struct {
```

Comment:
No, it has to stay, because `http.Server` calls handler even if we close the respective connection in the `ConnState` callback.